llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2430,6 +2430,30 @@ static bool isAllocSiteRemovable(Instruction *AI,
         unsigned OtherIndex = (ICI->getOperand(0) == PI) ? 1 : 0;
         if (!isNeverEqualToUnescapedAlloc(ICI->getOperand(OtherIndex), TLI, AI))
           return false;
+        // Do not fold compares to aligned_alloc calls, as they may have to
+        // return null in case the required alignment cannot be satisfied,
+        // unless we can prove that both alignment and size are valid.
+        if ((getAllocFnKind(AI) & AllocFnKind::Aligned) !=
+            AllocFnKind::Unknown) {
```

Comment:
Do we still need this check if we're also checking TLI? (Move the TLI check up here?)

Reply:
Not really, removed them